### PR TITLE
add kernel check in hyper

### PR DESCRIPTION
**Pull request by allencloud**
add kernel check in hyper 
to make only linux kernel 3.8.0+ can support hyper daemon

Signed-off-by: allencloud <allen.sun@daocloud.io>

### CODE REVIEW

File: daemon/daemon.go [hunk 1]
```diff
@@ -409,7 +412,21 @@ func (daemon *Daemon) Shutdown() error {
...
+		Kernel: k,
+		Major:  major,
+		Minor:  minor,
+	}
+
+	if v, err := kernel.GetKernelVersion(); err != nil {
+		return err
+	}
+
+	if kernel.CompareKernelVersion(*v, leastVersionInfo) < 0 {
```
Comment (gnawux) [inline daemon/daemon.go]: v is not defined, scope problem

File: daemon/daemon.go [hunk 2]
```diff
@@ -409,7 +412,21 @@ func (daemon *Daemon) Shutdown() error {
...
+		msg := fmt.Sprintf("Your Linux kernel(%d.%d.%d) is too old to support Hyper daemon(%d.%d.%d+)",
+			v.Kernel, v.Major, v.Minor, k, major, minor)
+		return fmt.Error(msg)
```
Comment (resouer) [inline daemon/daemon.go]: Should be `Errorf`, CI is also complaining this.

**Comment by allencloud** [comment]
Thanks for your careful code review. @gnawux @resouer 
Maybe I need to learn more details about how to run CI on my local machine.

**Comment by resouer** [comment]
Just checkout https://github.com/hyperhq/hyper/blob/master/.travis.yml

**Pull request merged**